src/Storefront/Resources/app/storefront/src/plugin/AGENTS.md

```diff
@@ -0,0 +1,153 @@
+
+## Plugin Structure
+
+```javascript
+import Plugin from 'src/plugin-system/plugin.class';
+
+export default class MyFeaturePlugin extends Plugin {
+    static options = { url: '', selector: '.element' };
+
+    init() { this._registerEvents(); }
+    update() { /* Re-initialize after DOM changes */ }
+    destroy() { /* Cleanup when plugin is destroyed */ }
+}
+```
+
+**Base Class Reference**: `Plugin` (imported) is the same as `window.PluginBaseClass` (global reference)
+- Preferred: Use ES6 import pattern shown above
+- Legacy: `class MyPlugin extends window.PluginBaseClass {}`
+
+### Instance Properties
+
+| Property           | Type                 | Purpose                        |
+|--------------------|----------------------|--------------------------------|
+| `this.el`          | `HTMLElement`        | DOM element plugin is bound to |
+| `this.$emitter`    | `NativeEventEmitter` | Event pub/sub on element       |
+| `this.options`     | `Object`             | Merged configuration options   |
+| `this._pluginName` | `String`             | Plugin name (kebab-case)       |
+
+## Registration
+
+Register in `main.js`:
+
+```javascript
+// Sync
+window.PluginManager.register('MyFeature', MyFeaturePlugin, '[data-my-feature]');
+
+// Async (preferred)
+window.PluginManager.register('MyFeature', () => import('./plugin/my-feature/my-feature.plugin'), '[data-my-feature]');
+```
+
+## Initialization
+
+**Data Attribute**: `data-{plugin-name}="true"` (kebab-case)
+
+```html
+<div data-my-feature="true"></div>
+```
+
+### Naming Convention
+
+Plugin class names are converted to kebab-case data attributes:
+
+1. Strip "Plugin" suffix
+2. Convert PascalCase to kebab-case
+3. Prefix with `data-`
+
+**Examples**:
+- `FilterPlugin` → `data-filter`
+- `VariantSwitchPlugin` → `data-variant-switch`
+- `MyCustomFeaturePlugin` → `data-my-custom-feature`
+- `AjaxModalPlugin` → `data-ajax-modal`
+
+### Data Attribute Values
+
+CSS selector `[data-my-feature]` matches by attribute presence, not value. Use `="true"` by convention. Remove attribute entirely to prevent initialization.
+
+## Options
+
+**Pass from template**:
+```html
+<div data-my-feature="true" data-my-feature-options='{"url": "/api"}'></div>
+```
+
+**Merge priority** (highest last):
+1. `static options` in class
+2. `PluginManager.register()` options
+3. `data-{plugin-name}-config` attribute (named config from PluginConfigManager)
+4. `data-{plugin-name}-options` attribute (inline JSON)
+5. `initializePlugin()` options
+
+## Events
+
+### Publishing Events
+
+```javascript
+// Element-scoped event (preferred for plugin-specific events)
+this.$emitter.publish('onClick', { data: 'payload' });
+
+// Global event (for cross-plugin/app-wide communication)
+document.$emitter.publish('globalEvent', { data: 'payload' });
+```
+
+### Subscribing to Events
+
+```javascript
+// Subscribe to element-scoped event
+element.addEventListener('onClick', (event) => console.log(event.detail));
```

Comment:
You would subscribe to a $emitter listener from a plugin by first getting the instance of the plugin. And then subscribe on the instance.

I think the `element.addEventListener` is wrong here.